### PR TITLE
Add CMake to most GitHub Actions builds using autobuild

### PR DESCRIPTION
**Pull request by ClaytonCalabrese**
Convert the GHA CMake builds to use autobuild rather than run_ci_tests.pl.

Add more CMake builds to GHA.

### CODE REVIEW

**Comment by iguessthislldo** [comment]
I'm still trying to look over this, but the `inst_u18_o1d0v1_xer0_j12` artifact for that last run has scoreboard pages with no content in them and the is missing a `output.log` file.

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -257,14 +264,48 @@ jobs:
...
+        </configuration>
+        <command name="log" options="on"/>
+        <command name="print_os_version"/>
+        <command name="print_env_vars"/>
+        <command name="print_ace_config"/>
+        <command name="print_make_version"/>
+        <command name="check_compiler" options="gcc"/>
+        <command name="print_perl_version"/>
+        <command name="print_autobuild_config"/>
+        <command name="auto_run_tests" options="script_path=tests dir=$GITHUB_WORKSPACE/OpenDDS --no-dcps -Config RTPS -Config LINUX -Config CXX11 -Config GH_ACTIONS --cmake"/>
```
Comment (iguessthislldo) [inline .github/workflows/build_and_test.yml:293]:
```suggestion
        <command name="auto_run_tests" options="script_path=tests dir=$GITHUB_WORKSPACE/OpenDDS --no-dcps --cmake"/>
```

`-Config` args aren't used to decide what CMake tests to run, so they can be removed if we're not running normal tests. Same for the other places.

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -1054,13 +1099,48 @@ jobs:
...
+        <variable name="log_root" value="$DDS_ROOT/${{ github.job }}_autobuild_workspace"/>
+        <variable name="project_root" value="$DDS_ROOT"/>
+        <variable name="root" value="$DDS_ROOT/${{ github.job }}_autobuild_workspace"/>
+        <variable name="junit_xml_output" value="Tests"/>
+        </configuration>
+        <command name="log" options="on"/>
+        <command name="print_os_version"/>
+        <command name="print_env_vars"/>
+        <command name="print_ace_config"/>
+        <command name="print_make_version"/>
```
Comment (iguessthislldo) [inline .github/workflows/build_and_test.yml:1124]:
```suggestion
        <command name="print_cmake_version"/>
        <command name="print_make_version"/>
```

I forgot to do this when I added CMake to the actions before, but could you do this for the places where I added `--cmake` to auto_run_tests as well as the places your adding.

**Review by iguessthislldo — CHANGES_REQUESTED** [review]
Can you also delete `run_ci_tests.pl` since we won't need it anymore?

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -257,14 +264,49 @@ jobs:
...
+        <command name="log" options="on"/>
+        <command name="print_os_version"/>
+        <command name="print_env_vars"/>
+        <command name="print_ace_config"/>
+        <command name="print_cmake_version"/>
+        <command name="print_make_version"/>
+        <command name="check_compiler" options="gcc"/>
+        <command name="print_perl_version"/>
+        <command name="print_autobuild_config"/>
+        <command name="auto_run_tests" options="script_path=tests dir=$GITHUB_WORKSPACE/OpenDDS --no-dcps -Config RTPS -Config LINUX -Config CXX11 -Config GH_ACTIONS --cmake"/>
```
Comment (iguessthislldo) [inline .github/workflows/build_and_test.yml:294]:
```suggestion
        <command name="auto_run_tests" options="script_path=tests dir=$GITHUB_WORKSPACE/OpenDDS --no-dcps --cmake"/>
```

File: .github/workflows/build_and_test.yml [hunk 4]
```diff
@@ -5678,16 +6191,137 @@ jobs:
...
+        <variable name="root" value="$FS_GHW/OpenDDS/${{ github.job }}_autobuild_workspace"/>
+        <variable name="junit_xml_output" value="Tests"/>
+        </configuration>
+        <command name="log" options="on"/>
+        <command name="print_os_version"/>
+        <command name="print_env_vars"/>
+        <command name="print_ace_config"/>
+        <command name="print_cmake_version"/>
+        <command name="print_make_version"/>
+        <command name="check_compiler" options="gcc"/>
```
Comment (iguessthislldo) [inline .github/workflows/build_and_test.yml:6299]: Do we want to make these correct? "gcc" on Windows

File: tests/cmake/CMakeLists.txt [hunk 5]
```diff
@@ -12,12 +12,14 @@ add_subdirectory(
...
-    add_subdirectory(C++11_typecode)
+  if(OPENDDS_OWNERSHIP_PROFILE)
+    add_subdirectory(Messenger)
+    if(NOT OPENDDS_STATIC)
+      add_subdirectory(generated_global)
+    endif()
+    if(NOT OPENDDS_SAFETY_PROFILE AND OPENDDS_CXX11)
+      add_subdirectory(C++11_typecode)
+    endif()
   endif()
```
Comment (iguessthislldo) [inline tests/cmake/CMakeLists.txt:23]:
```suggestion
  endif()
  if(NOT OPENDDS_SAFETY_PROFILE AND OPENDDS_CXX11)
    add_subdirectory(C++11_typecode)
  endif()
```

As far as I can tell typecode doesn't need ownership.

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

File: tests/cmake/CMakeLists.txt (hunk 5 above)
Comment (ClaytonCalabrese) [reply tests/cmake/CMakeLists.txt:23]: I think my brain was crosswired and saw this as C++11_Messenger. I'll fix it.

File: .github/workflows/build_and_test.yml (hunk 4 above)
Comment (ClaytonCalabrese) [reply .github/workflows/build_and_test.yml:6299]: I changed them to msvc for the windows build since it looks like that is what autobuild expects.

**Review by iguessthislldo — APPROVED** [review]
Looks good to me when ever the CI passes.

**Comment by mitza-oci** [comment]
> Looks good to me when ever the CI passes.

- [ ] resolve merge conflict

- [ ] cmake_w19_re_p1_stat_FM-08 has link errors

**Comment by mitza-oci** [comment]
>     * [ ]  cmake_w19_re_p1_stat_FM-08 has link errors

Is this an issue with `BUILD_SHARED_LIBS` in CMake?

**Pull request merged**